### ISSUE TITLE
When selecting Mumbai location - checking for Ipv6 shows France

### ISSUE THREAD

**Issue by kumar041** [AdguardTeam/AdGuardVPNForAndroid#228]
Hi,

I am using your vpn app(v 2.2.1) in my oneplus 6t running Android 13 with november security patch, side by side I am using adguard as well, so my problem is when I connected to Indian vpn server, my ipv4 address is mumbai, but my ipv6 address is france, so google and others mostly read my ipv6 address so they display everything in french. I having this peculiar kind of problem, I have uninstall and reinstall your app bit still same problem. Kindly look into it and correct the problem.

Google detection
![Screenshot_20221205-125740_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205578346-d2a09e20-86e8-464a-9e9e-9c6b5c16a815.png)

My ip address after connected to mumbai vpn

![Screenshot_20221205-125804_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205578545-8144fa75-b0bc-4d25-a69a-6497e43c15e7.png)

**Comment by maxikuzmin** [comment]
@kumar041 if you change location will your google language change from French?

**Comment by kumar041** [comment]
I am trying to explain is when I connected to Indian server, I am getting two different ip address, one is indian ip address that is ip address v4. Second one is French ip address that is ip address v6. The problem is most of the website reads ipv6 address that starts 2a04: something, this address is belongs to France. The vpn service provider M247 is allocating French ipv6 address.

I am uploading it again, kindly check and advice me how to rectify it, when I connect to Mumbai server my ipv4 and ipv6 address should be Indian ips. But M247 server is not allocating both ip address belongs to India.

![Screenshot_20221205-224244_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205700756-90e1f22d-51cd-4ff0-af9f-b1652860abff.png)

**Comment by maxikuzmin** [comment]
@kumar041 I see, but sometimes services may not show information correctly. Have you checked with MaxMind? You not answered, if you change location will your google language change from French?

**Comment by kumar041** [comment]
> @kumar041 I see, but sometimes services may not show information correctly. Have you checked with MaxMind? You not answered, if you change location will your google language change from French?

Yes, along with server location language changes in google. Ok maxmind says ipv6 belongs to mumbai, India. I do not understand it. But still google tells me, I am french, this what I am trying to say.

![Screenshot_20221206-182823_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205919293-566d1b84-9a4f-40f4-a62c-26c705922d0f.png)

![Screenshot_20221206-183024_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205919538-5a6d6425-658a-468f-b92c-c261422ca2a8.png)

![Screenshot_20221206-183039_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205919581-b62e75da-bc51-4824-9e72-a63d4d0faf01.png)

**Comment by maxikuzmin** [comment]
@kumar041 try to check the settings of your device, browser, Google account. Maybe you have set France in some of the settings

**Comment by kumar041** [comment]
> @kumar041 I see, but sometimes services may not show information correctly. Have you checked with MaxMind? You not answered, if you change location will your google language change from French?

Yes, along with server location language changes in google. Ok maxmind says ipv6 belongs to mumbai, India. I do not understand it. But still google tells me, I am french, this what I am trying to say.

![Screenshot_20221206-182823_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205919293-566d1b84-9a4f-40f4-a62c-26c705922d0f.png)

![Screenshot_20221206-183024_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205919538-5a6d6425-658a-468f-b92c-c261422ca2a8.png)

![Screenshot_20221206-183039_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/205919581-b62e75da-bc51-4824-9e72-a63d4d0faf01.png)


Every thing in my phone set to India, I have checked twice, there is no change in it. So your vpn still doing some error, it is vpn ip address error. So vpn provider M247 have to do something about this.

**Comment by maxikuzmin** [comment]
@kumar041 try disabling ipv6 in low level settings. Does it take French out of Google?

**Comment by kumar041** [comment]
> @kumar041 try disabling ipv6 in low level settings. Does it take French out of Google?

Hi, 

I have disabled the ipv6 in low level as per your instruction, but still google welcomes me in french. Kindly check it.

Ipv6 disabled

![Screenshot_20221207-144012_AdGuard VPN.png](https://user-images.githubusercontent.com/97349871/206138058-f032f7ee-a42a-4db6-ba9f-5d7d9a9595e5.png)

Connected to indian server

![Screenshot_20221207-144423_AdGuard VPN.png](https://user-images.githubusercontent.com/97349871/206138240-44d63bd8-3310-4d6f-b526-4a920fe8921f.png)

Google in french

![Screenshot_20221207-144408_Vivaldi Browser Snapshot.png](https://user-images.githubusercontent.com/97349871/206138402-9ad94adc-7643-4066-87ce-697aafbd3b26.png)

**Comment by maxikuzmin** [comment]
@kumar041 I don't have this repeated. Try clicking on the globe (like in your screenshot) and change the language. Also go into your Google account settings and check what you have in "Language and Region" - check "Search Language" and "Search Region". 
I think you have French in "Search Language" and that's the problem

**Comment by kumar041** [comment]
> @kumar041 I don't have this repeated. Try clicking on the globe (like in your screenshot) and change the language. Also go into your Google account settings and check what you have in "Language and Region" - check "Search Language" and "Search Region". 
> I think you have French in "Search Language" and that's the problem

I have checked twice in my mobile as well as logged through browser on my pc, both says language is english (us) and checked the region is set to india, i can see it in my playstore also. So there is no problem in my side. If you are talking about web browser setting I never log into browser due to security reason. I know I can change language and search setting on google search page. I think everytime I say I have a problem, adguard support overall not providing any solution. Just everything is my problem. Even when I connect to the indian server after four hours server reset the connection on its own, it says there is a problem and vpn app can't connect to server. I have manually close the vpn app and restart it or sometime I have to wait for sometime like five minutes to reconnect. From evening onwards I think indian server ping time is 236ms, England server is 147ms. So something is going on again. Noe I am using my pc, I am connected to vpn on french server, youtube and google says I am from England. So overall everything fine on your side, my mobile is having problem. Am I correct.

**Comment by maxikuzmin** [comment]
@kumar041 if you have a problem with connection dropping, please start a new issue, as it is not relevant to this topic. 
So, what we have: 

- Initially you said that the problem is in ipv6, because on ipv6 in one of the resources displays France. 
- You checked the location of VPN on maxmind site, and it does not display France, which is considered more reliable. 
- Next you disabled your ipv6, but you still have French displayed. So the problem is not with ipv6 or VPN.

I repeated your steps, and I don't have this error again. When I change locations, my language does not change in Google. As far as I know, the language in Google is displayed based on the settings you set in the system. Google automatically pulls the language from the system, but you can also manually select a different language in the Google preferences by going to Settings -> Language & Region

**Comment by kumar041** [comment]
> @kumar041 if you have a problem with connection dropping, please start a new issue, as it is not relevant to this topic. 
> So, what we have: 
> 
> - Initially you said that the problem is in ipv6, because on ipv6 in one of the resources displays France. 
> - You checked the location of VPN on maxmind site, and it does not display France, which is considered more reliable. 
> - Next you disabled your ipv6, but you still have French displayed. So the problem is not with ipv6 or VPN.
> 
> I repeated your steps, and I don't have this error again. When I change locations, my language does not change in Google. As far as I know, the language in Google is displayed based on the settings you set in the system. Google automatically pulls the language from the system, but you can also manually select a different language in the Google preferences by going to Settings -> Language & Region
> 
> 

Hi,

As usual  I am wrong you are right. And close the case. There is nothing to say more about it. Good bye.

**Comment by maxikuzmin** [comment]
@kumar041 I'm not refusing to help you, we've dealt with this issue in depth. The outcome of this case I described above. The language does not change because of the VPN, you have tested this case without ipv6, and the situation is the same

**Comment by kumar041** [comment]
> @kumar041 I'm not refusing to help you, we've dealt with this issue in depth. The outcome of this case I described above. The language does not change because of the VPN, you have tested this case without ipv6, and the situation is the same

Hi,

Kindly do not misunderstood me, I did not got upset or got angry over this, I am satisfied with your answer and appreciate your effort to solve it. I just thought about this problem in different way, we can't change google and other website like what we want them to do, so they can't read the ip address correctly, that is not your vpn or my problem, so I decided to lets leave it as it is and lets move on that's all.
So that I got satisfied on this case, I request to close it.

Thank you,
Rajkumar

**Comment by kumar041** [comment]
> @kumar041 I'm not refusing to help you, we've dealt with this issue in depth. The outcome of this case I described above. The language does not change because of the VPN, you have tested this case without ipv6, and the situation is the same

Hi,

Kindly do not misunderstood me, I did not got upset or got angry over this, I am satisfied with your answer and appreciate your effort to solve it. I just thought about this problem in different way, we can't change google and other website like what we want them to do, so they can't read the ip address correctly, that is not your vpn or my problem, so I decided to lets leave it as it is and lets move on that's all.
So that I got satisfied on this case, I request to close it.

Thank you,
Rajkumar
